scene/animation/animation_player.cpp

```diff
@@ -1072,25 +1072,31 @@ void AnimationPlayer::_animation_process2(double p_delta, bool p_started) {
 	accum_pass++;
 
 	bool seeked = c.seeked; // The animation may be changed during process, so it is safer that the state is changed before process.
+	float blend = 1.0; // First animation we play at 100% blend
+
 	if (p_delta != 0) {
 		c.seeked = false;
 	}
 
-	_animation_process_data(c.current, p_delta, 1.0f, seeked, p_started);
-
-	List<Blend>::Element *prev = nullptr;
-	for (List<Blend>::Element *E = c.blend.back(); E; E = prev) {
+	List<Blend>::Element *next = NULL;
+	for (List<Blend>::Element *E = c.blend.front(); E; E = next) {
 		Blend &b = E->get();
-		float blend = b.blend_left / b.blend_time;
+		_animation_process_data(b.data, p_delta, blend, false, false);
+		blend = 1.0 - b.blend_left / b.blend_time; // This is how much to blend the NEXT animation
 		b.blend_left -= Math::absf(speed_scale * p_delta);
-		prev = E->prev();
+		next = E->next();
 		if (b.blend_left < 0) {
-			c.blend.erase(E);
+			// If the blend of this has finished, we need to remove ALL the previous blends
+			List<Blend>::Element *prev;
+			while (E) {
+				prev = E->prev();
+				c.blend.erase(E);
+				E = prev;
+			}
 		}
-		// The effect of animation changes during blending is unknown...
-		// In that case, we recommends to use method call mode "deferred", not "immediate".
```

Comment:
Do not delete this comment. we cannot guarantee the behavior if the animation is changed by MethodTrack in Immediate mode.

Reply:
How can I test this?  I hoped it was fixed, and the line the comment was on was removed.

Reply:
This is probably impossible to avoid, since if an animation is changed in immediate mode while iterating on a blend, the subsequent iterations should be discarded. See also #69685 and #69722.

Reply:
So just to clarify:

- One of the animations playing in the blend has an event in it that triggers a new animation to start playing.
- We're already looping the list of blends.
- A new blend would be added to that list.
- ??? unknown happens?

If it gets added to the end of the list, it should process after the other blends and simply go on its way.  I don't think this is a problem with the order it's processed now, but I'd need to verify it.  It's possible it would be skipped that frame and not start until the next frame, but I don't feel like that's a notable issue.

It is possible that c.blend.clear() I added could clear out the list of blends if we try to blend in with a custom time of 0, so that could be a problem.

Reply:
See MRP in #69722 for reference. There is no blend there, but if you add a blend, I guess the problem will show up.

There are animations A, B, and C. A and B are now blended. In this case, the iterate order is A->B. If the method track in A changes A to C, and the method callback mode is Immediate, then B will not be processed in the same frame; if that mode is Deferred, then it should be processed.

Reply:
So I'm trying to test this, but I can't seem to add animation events:
![image](https://user-images.githubusercontent.com/7751182/209016867-18db802a-21b3-4316-abd8-0ca0bfb79d64.png)

I changed the import settings the way it said to:
![image](https://user-images.githubusercontent.com/7751182/209016934-7ef2d523-0740-42ad-a5a7-acdf83c5943f.png)

Reply:
You can make unique the animationlib, then you will be able to edit the imported animation.
![image](https://user-images.githubusercontent.com/61938263/209017437-4edbbef0-8b67-4eef-97f1-6b04be0242e7.png)